twisted/web/test/test_http2.py

```diff
@@ -1403,6 +1403,32 @@ def validate(streamID):
         return a._streamCleanupCallbacks[1].addCallback(validate)
 
 
+    def test_resetAfterBody(self):
+        """
+        A client that immediately resets after sending the body causes Twisted
+        to send no response.
+        """
+        f = FrameFactory()
+        transport = StringTransport()
+        a = H2Connection()
+        a.requestFactory = DummyHTTPHandler
+
+        requestBytes = f.preamble()
```

Comment:
You need it because all HTTP/2 connections begin with the clients sending the client preamble as defined in [RFC 7540](https://tools.ietf.org/html/rfc7540#section-3.5). This is a fixed byte stream (`b"PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n"`) that is required to begin speaking HTTP/2.